### PR TITLE
User interface Complete

### PR DESCRIPTION
**Pull request by Aaron-RN**
Displays board information and gets input from player to determine where to play the symbol

### CODE REVIEW

**Review by bolah2009 — CHANGES_REQUESTED** [review]
I believe this is milestone one, right?

Follow the steps in the [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)

> ## Milestone 1: Project setup
> 
> Setup project as it is described in requirements.
> 
> In addition, create 2 main directories:
> - lib - for the game logic (HINT: as this directory is empty you can use .gitkeep file)
> - bin - with the [executable file] https://commandercoriander.net/blog/2013/02/16/making-a-ruby-script-executable/) called main.rb.
> - main.rb file should be responsible for user interaction. This is the only one place where puts and get.chomp can be used.
> Make your main.rb file displaying “Hello world!” when you use bin/main.rb as a bash command.

Let's make the `lib` directories visible and name the `branch` `project_setup` or anything similar, this current branch name is for **Milestone 2** not one.


#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Please, let's submit milestone 1 and 2 with a different pull request.

File: bin/main [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+def getPlayerInput
+  #puts "Player #{$PlayerTurn.symbol}'s Turn'"
+  print "Choose an available position to play: "
+  posChosen = gets.chomp.to_i
+  puts "\n"
+  #checkIfInteger(posChosen)
+  #posToBoard(1, $PlayerTurn.symbol, "red") if checkBoardPos(posChosen)
+  posToBoard(posChosen, "X", "green")
+end
+
```
Comment (samgaco) [inline bin/main:138]: - [x] A loop should be declared (white, until, etc) that executes the game to the console and doesn't terminate until it finds a result (winning player / draw)

File: bin/main [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+def posToBoard(pos, sym, color)
+  $Board[pos-1] = "#{sym}".green if color == "green"
+  $Board[pos-1] = "#{sym}".red if color == "red"
+  $Board[pos-1] = "#{sym}".blue if color == "blue"
+  displayBoard()
+end
+
+def getPlayerInput
+  #puts "Player #{$PlayerTurn.symbol}'s Turn'"
+  print "Choose an available position to play: "
```
Comment (samgaco) [inline bin/main]: - [x] Can we see the available positions to choose from? If we input an invalid option, will the game warn us? Those things should be implemented

File: bin/main [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    $Board << i
+  end
+end
+
+initializeBoard()
+
+def displayBoard
+  for i in 0..(BOARD_COLS*BOARD_ROWS)+((BOARD_ROWS-1)*BOARD_COLS)-1
+      if i < BOARD_COLS
+        print " #{$Board[i]}" unless i == BOARD_COLS-1
```
Comment (samgaco) [inline bin/main:45]: - [x] The game board should be displayed after each move

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @Aaron-RN :high_brightness: 

I have indicated some corrections. It would be good if the mock **[optional]** could be shown in the console as well, along with the **required** corrections I have pointed you out in the comments below.

Also:

- [x] Branch conflicts should be solved


Regards

-----------------------------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: bin/main (hunk 3 above)
Comment (Aaron-RN) [reply bin/main:45]: Hey yeah, we did all of the aforementioned changes already in the Game Logic branch...
We only created the display functions here in the user_interface branch

Comment (Aaron-RN) [reply bin/main:45]: All of our other branches have been approved

**Review by Ispirett — CHANGES_REQUESTED** [review]
Hey, @Aaron-RN @zhuchini  great work so far.  :100: 
This project was summited for milestone 2.
Think of milestone 2 as an MVP (minimal viable product) 
So you don't need everything workin perfectly but you have to something working. 

**Things to ensure working before you resubmit.**
[OPTIONAL] game ask players for their names

 [REQUIRED] game tells which player turn it is

 [REQUIRED] game asks player to select from available moves

 [REQUIRED] game informs player if selected move is invalid

 [REQUIRED] game displays board after player move

 [REQUIRED] game informs player if selected move is a winning move

 [REQUIRED] game informs player if selected move is a draw move

 [REQUIRED] game repeats all actions for next player's move

You can puts `"some message"` as a placeholder for code.

**currently your `main` file is missing the .rb extension please rename to `main.rb `**

**Comment by Aaron-RN** [comment]
> Hey, @Aaron-RN @zhuchini great work so far. 💯
> This project was summited for milestone 2.
> Think of milestone 2 as an MVP (minimal viable product)
> So you don't need everything workin perfectly but you have to something working.
> 
> **Things to ensure working before you resubmit.**
> [OPTIONAL] game ask players for their names
> 
> [REQUIRED] game tells which player turn it is
> 
> [REQUIRED] game asks player to select from available moves
> 
> [REQUIRED] game informs player if selected move is invalid
> 
> [REQUIRED] game displays board after player move
> 
> [REQUIRED] game informs player if selected move is a winning move
> 
> [REQUIRED] game informs player if selected move is a draw move
> 
> [REQUIRED] game repeats all actions for next player's move
> 
> You can puts `"some message"` as a placeholder for code.
> 
> **currently your `main` file is missing the .rb extension please rename to `main.rb `**

So you want us to copy the game_logic branch and rename user_interface branch?
Also the file no longer has the ,rb extension because it was a requirement for milestone 1: 
https://commandercoriander.net/blog/2013/02/16/making-a-ruby-script-executable/

**Comment by Aaron-RN** [comment]
> Hey, @Aaron-RN @zhuchini great work so far. 💯
> This project was summited for milestone 2.
> Think of milestone 2 as an MVP (minimal viable product)
> So you don't need everything workin perfectly but you have to something working.
> 
> **Things to ensure working before you resubmit.**
> [OPTIONAL] game ask players for their names
> 
> [REQUIRED] game tells which player turn it is
> 
> [REQUIRED] game asks player to select from available moves
> 
> [REQUIRED] game informs player if selected move is invalid
> 
> [REQUIRED] game displays board after player move
> 
> [REQUIRED] game informs player if selected move is a winning move
> 
> [REQUIRED] game informs player if selected move is a draw move
> 
> [REQUIRED] game repeats all actions for next player's move
> 
> You can puts `"some message"` as a placeholder for code.
> 
> **currently your `main` file is missing the .rb extension please rename to `main.rb `**

A few of those requirements are related to game_logic and also the guidelines mentioned on Milestone 2 explicitly said that the game doesn't "have to work yet"

**Review by Ispirett — CHANGES_REQUESTED** [review]
Hey @Aaron-RN  really good job with the changes.

When running the command ` ruby bin/main.rb`.

I'm am presented with 
`ruby bin/main.rb
Traceback (most recent call last):
ruby: No such file or directory -- lib/main.rb (LoadError)`

please rename the file from `main` to `main.rb`.

Also yes you don't need everything working in this milestone.
just the required ones I listed before.

The  last thing to implement is player win check and draw a check
nothing perfect or fancy but just a message that informs the players if there is a win or draw.

**Comment by Aaron-RN** [comment]
> Hey @Aaron-RN really good job with the changes.
> 
> When running the command ` ruby bin/main.rb`.
> 
> I'm am presented with
> `ruby bin/main.rb Traceback (most recent call last): ruby: No such file or directory -- lib/main.rb (LoadError)`
> 
> please rename the file from `main` to `main.rb`.
> 
> Also yes you don't need everything working in this milestone.
> just the required ones I listed before.
> 
> The last thing to implement is player win check and draw a check
> nothing perfect or fancy but just a message that informs the players if there is a win or draw.

You don't run it by typing main.rb you run it by typing main in the terminal..
as per requested by Milestone 1: making a ruby file executable: https://commandercoriander.net/blog/2013/02/16/making-a-ruby-script-executable/

Also How do I run a Game won or Game Draw message without game_logic?

**Review by whiz25 — APPROVED** [review]
## Good job :clap: 
Your project meets all the requirements for this milestone. Go ahead and merge.

#### Marin Afani

**Pull request merged**